meson_init.sh

```diff
@@ -120,12 +126,14 @@ if [[ ! -z "${ARG_tock_dir}" ]]; then
   fi
 fi
 
-readonly DEFAULT_RISCV_TOOLS=/tools/riscv
-TOOLCHAIN_PATH="${TOOLCHAIN_PATH:-$DEFAULT_RISCV_TOOLS}"
-CROSS_FILE=$(mktemp /tmp/toolchain.XXXXXX.txt)
-cp toolchain.txt "$CROSS_FILE"
-perl -pi -e "s#$DEFAULT_RISCV_TOOLS#$TOOLCHAIN_PATH#g" "$CROSS_FILE"
-echo "Set up toolchain file at $CROSS_FILE." >&2
+if [[ -f "${ARG_toolchain_file}" ]]; then
+  echo "Using meson toolchain file at $ARG_toolchain_file." >&2
+else
+  CROSS_FILE=$(mktemp /tmp/toolchain.XXXXXX.txt)
+  cp toolchain.txt "$CROSS_FILE"
+  perl -pi -e "s#$DEFAULT_RISCV_TOOLS#$TOOLCHAIN_PATH#g" "$CROSS_FILE"
+  echo "Set up toolchain file at $CROSS_FILE." >&2
+fi
```

Comment:
If the user manually specified the cross-file and it cannot be found we should complain, rather than going to the else branch here and having meson fail.

Reply:
Yeah that's fair. Hopefully I'm tracking this better now, by tracking if it's user-specified or not.